### PR TITLE
Fix "Good" collision not honoring mesh groups in OBJ files

### PR DESCRIPTION
**Pull request by sabrina-shanman**
https://highfidelity.manuscript.com/f/cases/4828/Compound-hulls-being-treated-like-dynamic-basic-hulls

A bug in the way models were loaded was causing OBJ-based models to not be broken down into meshes when the "Good" collision type was selected in the create model dialogue. The fix in this PR is to store the collision model separately from the rendered model in that case, since the rendered model has meshes combined for optimization.

This is a "conservative" fix. One of the issues with model management this PR does not address is that the model is loaded twice if it needs to be stored in a separate resource variable for collision purposes. There are also issues with the resource pipeline in general that we'd like to address in the future.

It's also worth noting that this PR does not add any "smart" handling of the OBJ model files. Meshes with the same material will only be separated if OBJ file divides them into different groups.

## TEST PLAN
- Verify this PR improves the collision bounds of [this test model](https://gist.githubusercontent.com/sabrina-shanman/457b40f2a66b517b0dc41196bf6ab96e/raw/bf5405a23cefe3662e30d14edf4d3386a406e128/wamConsole_hull_2_grouptest.obj) compared to master using these steps:
  - In a domain with create rights, open the create app.
  - In the create tools menu, under the create tab, select the "Model" icon to open the "New Model Entity" dialogue
  - Paste the URL of the provided test model into the "Model URL" text field. Select "Good - Sub-Meshes" from the "Automatic Collisions" drop-down menu.
  - Click "Add" to add the test model to the scene. Re-size the model so it is roughly the size of your avatar. The resized test model should look like a grey, chair-shaped object.
  - Walk into the test model from various directions and stand on top of it. Rotate the model so your avatar can try standing on the side and the bottom as well.
  - In this PR, the collision of the avatar with the model should look relatively realistic. In particular, the avatar should be able to stand on the "seat" of the chair-shaped model. In master, the avatar will be unable to reach the inner edge of the model.
- In this PR and in master, visit various domains and compare the way the avatar collides with things in the domain. The collision should be the same in most cases, and possibly more accurate in rare cases.
  - You can observe the collision shapes directly by enabling the debug collision meshes (however, sometimes they are hard to see because the visible models overlap them). You can make collision meshes visible by enabling Developer->Physics->Show Bullet Collision (prerequisite: Settings->Developer Menu is enabled)

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13701/HighFidelity-Beta-PR14156-e15a41c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13701/HighFidelity-Beta-PR14156-e15a41c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13701/HighFidelity-Beta-Interface-PR14156-e15a41c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13701/HighFidelity-Beta-PR14156-e15a41c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13701/HighFidelity-Beta-Interface-PR14156-e15a41c.exe

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 1]
```diff
@@ -510,7 +514,15 @@ void RenderableModelEntityItem::computeShapeInfo(ShapeInfo& shapeInfo) {
...
-        auto& meshes = model->getGeometry()->getMeshes();
+        std::vector<std::shared_ptr<const graphics::Mesh>> compoundMeshes;
+        if (type == SHAPE_TYPE_SIMPLE_COMPOUND) {
+            auto& fbxMeshes = _compoundShapeResource->getFBXGeometry().meshes;
+            compoundMeshes.reserve(fbxMeshes.size());
+            for (auto& fbxMesh : fbxMeshes) {
+                compoundMeshes.push_back(fbxMesh._mesh);
+            }
+        }
+        auto& meshes = (type == SHAPE_TYPE_SIMPLE_COMPOUND ? compoundMeshes : model->getGeometry()->getMeshes());
```
Comment (AndrewMeadows) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: Avoid creating `compoundMeshes` for the case where it is never used.  Maybe this works instead?
```
        std::vector<std::shared_ptr<const graphics::Mesh>> meshes;
        if (type == SHAPE_TYPE_SIMPLE_COMPOUND) {
            auto& fbxMeshes = _compoundShapeResource->getFBXGeometry().meshes;
            meshes.reserve(fbxMeshes.size());
            for (auto& fbxMesh : fbxMeshes) {
                meshes.push_back(fbxMesh._mesh);
            }
        } else {
            meshes = model->getGeometry()->getMeshes();
        }
```

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13706/HighFidelity-Beta-PR14156-57f67ff.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13706/HighFidelity-Beta-PR14156-57f67ff.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13706/HighFidelity-Beta-Interface-PR14156-57f67ff.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13706/HighFidelity-Beta-PR14156-57f67ff.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13706/HighFidelity-Beta-Interface-PR14156-57f67ff.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/13953/HighFidelity-Beta-PR14156-57f67ff.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13953/HighFidelity-Beta-PR14156-57f67ff.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/13953/HighFidelity-Beta-Interface-PR14156-57f67ff.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13953/HighFidelity-Beta-PR14156-57f67ff.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/13953/HighFidelity-Beta-Interface-PR14156-57f67ff.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Pull request merged**